cmake/FindGzAssimp.cmake

```diff
@@ -15,22 +15,31 @@
 #
 ########################################
 # Find Assimp
+# Provides a GzAssimp alias to avoid conflicts with other packages
+# that might provide a FindAssimp function, such as dartsim
 
-include(GzPkgConfig)
+find_package(assimp CONFIG QUIET)
 
-if(ASSIMP_FIND_VERSION)
-  gz_pkg_check_modules_quiet(GzAssimp "assimp >= ${ASSIMP_FIND_VERSION}")
+set(GzAssimp_FOUND ${assimp_FOUND})
+if (MSVC)
+  set(GzAssimp_LIBRARIES assimp::assimp)
 else()
-  gz_pkg_check_modules_quiet(GzAssimp "assimp")
+  set(GzAssimp_LIBRARIES ${ASSIMP_LIBRARIES})
```

Comment:
Is the distinction between MSVC and others necessary here? Can't they all use the `assimp::assimp` target?

Reply:
I was hoping that would be the case but I just get very puzzling errors in Ubuntu if I use the `assimp::assimp` target, specifically:

```
luca@luxury:~/ws_garden$ colcon build --merge-install  --packages-select gz-cmake3 gz-common5
Starting >>> gz-cmake3
Finished <<< gz-cmake3 [0.22s]                  
Starting >>> gz-common5
--- stderr: gz-common5                         
CMake Warning (dev) at /usr/lib/x86_64-linux-gnu/cmake/assimp-5.0/assimpTargets.cmake:54 (if):
  if given arguments:

    "ON"

  An argument named "ON" appears in a conditional statement.  Policy CMP0012
  is not set: if() recognizes numbers and boolean constants.  Run "cmake
  --help-policy CMP0012" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.
Call Stack (most recent call first):
  /usr/lib/x86_64-linux-gnu/cmake/assimp-5.0/assimp-config.cmake:1 (include)
  /home/luca/ws_garden/install/share/cmake/gz-cmake3/cmake3/FindGzAssimp.cmake:21 (find_package)
  /home/luca/ws_garden/install/share/cmake/gz-cmake3/cmake3/GzUtils.cmake:231 (find_package)
  CMakeLists.txt:127 (gz_find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

Warning, assimp below 5.2.0 detected, setting compatibility mode
CMake Error in graphics/src/CMakeLists.txt:
  Imported target "assimp::assimp" includes non-existent path

    "/usr/lib/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in graphics/src/CMakeLists.txt:
  Imported target "assimp::assimp" includes non-existent path

    "/usr/lib/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Generate step failed.  Build files cannot be regenerated correctly.
make: *** [Makefile:1702: cmake_check_build_system] Error 1
---
Failed   <<< gz-common5 [4.43s, exited with code 2]

Summary: 1 package finished [5.23s]
  1 package failed: gz-common5
  1 package had stderr output: gz-common5
```

It seems, at least on Ubuntu 20.04, the target links to the wrong library path

Reply:
ha, got it, thanks